opensquirrel/MatrixExpander.py

```diff
@@ -5,7 +5,7 @@
 
 
 # This should only be used for testing and on circuits with low number of qubits.
-def getBigMatrix(semantic, qubitOperands, totalQubits):
+def getBigMatrix(semantic: MultiQubitMatrixSemantic, qubitOperands, totalQubits):
```

Comment:
The correct type here is the parent class `Semantic` in the current code

Reply:
(see below `isinstance`)

Reply:
Yes, makes sense!

Perhaps we should make Semantic an ABC then?

Reply:
It is indeed an ABC! I just only recently discovered that you can use this in Python with an extra module - it's not in the language by default I believe.

Reply:
You need to import the `abc` library indeed (https://docs.python.org/3/library/abc.html).

I was also thinking if it would be an idea to create ABCs for the various passes. And Users can then define new passes based on those interfaces. We have done a similar thing with NetSquid.